### PR TITLE
Update battery pool SoC calculation

### PR DESCRIPTION
**Pull request by matthias-wende-frequenz**
Fix the SoC calculation such that it never exceeds 100%.

### CODE REVIEW

File: src/frequenz/sdk/timeseries/battery_pool/_metric_calculator.py [hunk 1]
```diff
@@ -424,7 +424,7 @@ def calculate(
             soc_scaled = (
                 (soc - soc_lower_bound) / (soc_upper_bound - soc_lower_bound) * 100
             )
-            soc_scaled = max(soc_scaled, 0)
+            soc_scaled = min(max(soc_scaled, 0), 100)
```
Comment (daniel-zullo-frequenz) [inline src/frequenz/sdk/timeseries/battery_pool/_metric_calculator.py:428]: This is clamping `soc_scaled` from 0 to 100 and I understand it fixes the error. However I have the feeling the previous code and the current patch are hiding the fact that `soc` is not within the bounds to get a value lower than 0 or greater than 100. If that's the case I'd suggest to explicitly clamp the `soc` to the bounds instead.
Something around:
```python
soc = max(min(soc, soc_upper_bound), soc_lower_bound)
soc_scaled = (soc - soc_lower_bound) / (soc_upper_bound - soc_lower_bound) * 100
```
Though my suggestion does not protect us for cases where the bounds were not set correctly. So in that sense clamping from 0 to 100 is safer.

File: src/frequenz/sdk/timeseries/battery_pool/battery_pool.py [hunk 2]
```diff
@@ -353,10 +353,11 @@ def soc(self) -> MetricAggregator[Sample[Percentage]]:
         ```
         working_batteries: Set[BatteryData] # working batteries from the battery pool
 
-        soc_scaled = max(
+        # TODO what bound is used?
```
Comment (daniel-zullo-frequenz) [inline src/frequenz/sdk/timeseries/battery_pool/battery_pool.py]: do you want to keep this TODO message? just checking...

**Review by daniel-zullo-frequenz — COMMENTED** [review]
Other than a suggestion to consider and a question, LGTM
Also this might need an entry to the release notes as it is fixing the SoC metric.

**Comment by llucax** [comment]
Also release notes.

File: src/frequenz/sdk/timeseries/battery_pool/battery_pool.py (hunk 2 above)
Comment (matthias-wende-frequenz) [reply src/frequenz/sdk/timeseries/battery_pool/battery_pool.py]: no, classic oversight from my side :).

File: src/frequenz/sdk/timeseries/battery_pool/_metric_calculator.py (hunk 1 above)
Comment (daniel-zullo-frequenz) [reply src/frequenz/sdk/timeseries/battery_pool/_metric_calculator.py:428]: We agreed IRL to keep the current changes and add an inline comment to clarify `soc` might be out of bounds

File: src/frequenz/sdk/timeseries/battery_pool/battery_pool.py [hunk 3]
```diff
@@ -347,16 +347,17 @@ def consumption_power(self) -> FormulaEngine[Power]:
     def soc(self) -> MetricAggregator[Sample[Percentage]]:
         """Fetch the normalized average weighted-by-capacity SoC values for the pool.
 
-        The values are normalized to the 0-100% range.
+        The values are normalized to the 0-100% range and if clamped if the SoC is
```
Comment (daniel-zullo-frequenz) [inline src/frequenz/sdk/timeseries/battery_pool/battery_pool.py]:
```suggestion
        The values are normalized to the 0-100% range and clamped if the SoC is
```

**Review by daniel-zullo-frequenz — COMMENTED** [review]
A last cosmetic suggestion and LGTM

File: src/frequenz/sdk/timeseries/battery_pool/battery_pool.py (hunk 3 above)
Comment (matthias-wende-frequenz) [reply src/frequenz/sdk/timeseries/battery_pool/battery_pool.py]: fixed

**Review by daniel-zullo-frequenz — APPROVED** [review]

**Pull request merged**